storage/session.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package storage
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/bradfitz/gomemcache/memcache"
+	"github.com/eko/gocache/lib/v4/cache"
+	"github.com/eko/gocache/lib/v4/store"
+)
+
+var _ SessionStore = (*SessionStoreImpl[[]byte])(nil)
+
+// defaultSessionDataTTL is the default time to live for session data
+// some stores require a default
+var defaultSessionDataTTL = 15 * time.Minute
+
+// StringOrBytes is a type that can be either a string or a byte slice
+// used for generic type constraints
+type StringOrBytes interface {
+	~string | ~[]byte
+}
+
+// SessionStoreImpl is an implementation of the SessionStore interface
+// It handles logic for all session store types
+type SessionStoreImpl[T StringOrBytes] struct {
+	underlying *cache.Cache[T]
```

Comment:
does everything go through the memcache API now? (in-memory as well?)

If not, this should be named something like MemcachedSessionStore?

Reply:
This is a generic impl, memcached/redis and in-memory initialize this with a different underlying cache. The lib used is a facade for the different caches.